network/src/interface/mod.rs

```diff
@@ -304,7 +304,10 @@ where
                 }
             }
             _ => {
-                unreachable!("Unexpected notification received from Peer actor");
+                warn!(
+                    "Unexpected notification received from Peer actor: {:?}",
+                    notif
+                );
```

Comment:
can you actually use a `MUST_FIX` structured log? https://github.com/libra/libra/blob/b4061ca3443ed0c512c677a5a9ef755578b83d15/execution/executor/src/lib.rs#L189

Reply:
and I think error is better also (cc @gregnazario )

Reply:
(same for others)

Reply:
oh I realize that I'm too late : c

Reply:
@mimoo, no worries! If you feel strongly about the logging type and the MUST_FIX, feel free to create a PR that handles it and send it across to us. I went for the logging types recommended by @davidiw -- noting the log spamming issue with higher levels. Others might have other thoughts?

Reply:
so I think we need to create some guidance and tooling around these type of errors. My idea was that if we can create some detection rules on MUST_FIX logs, then we'll notice them faster in the sea of logging that we'll have to monitor.

Reply:
also if it's a MUST_FIX, we should worry about that rather than the spamming (unless the spamming leads to a DoS ofc)

Reply:
IMO, if it's a MUST_FIX, shouldn't we just fix it?

If we're worried about having an "unreachable" macro that logs when something unexpected happens, but not panic, we could build that into the logging system.

Reply:
yeah it would be cool to have a must_fix!() macro that just emits a slog of category "MUST_FIX"